dbms/src/Interpreters/Join.cpp

```diff
@@ -1768,7 +1862,19 @@ void Join::spillProbeSideBlocks(UInt64 part_id, Blocks && blocks)
     probe_spiller->spillBlocks(std::move(blocks), part_id);
 }
 
-void Join::spillMostMemoryUsedPartitionIfNeed()
+void Join::markBuildSideSpillData(UInt64 part_id, Blocks && blocks, size_t stream_index)
+{
+    if (!blocks.empty())
+        getBuildSideMarkdeSpillData(stream_index).emplace_back(part_id, std::move(blocks));
```

Comment:
assert that `getBuildSideMarkdeSpillData(stream_index).second` is empty before emplace_back?

Reply:
ok, added.